client/src/java/com/zimbra/client/ZFilterCondition.java

```diff
@@ -957,7 +956,12 @@ public ZDateCondition(DateOp op, String dateStr) throws ServiceException {
 
         ZDateCondition(FilterTest.DateTest test) throws ServiceException {
             this.op = DateOp.of(Sieve.DateComparison.fromString(test.getDateComparison()), test.isNegative());
-            this.date = new Date(test.getDate() * 1000);
+            //this.date = new Date(test.getDate() * 1000);
+            try {
+                this.date = new SimpleDateFormat("yyyyMMdd").parse(test.getDate());
```

Comment:
can we use a constant to hold the pattern

Reply:
not needed